src/jubilation-horizontal-bar/index.jsx

```diff
@@ -0,0 +1,124 @@
+// @flow
+import React from 'react';
+import uuid from 'uuid';
+import getDomainMap from '../helpers/scatter';
+import Rect from '../jubilation-rect';
+import Label from '../jubilation-label';
+import getContext from '../helpers/context';
+import getTicks from '../helpers/axis';
+import getTextWidth from '../helpers/text';
+
+type Props = {
+  data: number[],
+  labels: string[],
+  gapWidth: number,
+  color?: string,
+  style?: Object,
+};
+type Context = { JubilationContext: JubilationContext };
+
+export default class JubilationHorizontalBarChart extends React.Component {
+  props: Props
+  context: Context
+  uuid: string
+  domainMap: DomainMap
+
+  static defaultProps: Props = { data: [], labels: [], gapWidth: 0.05 };
+
+  static contextTypes = { JubilationContext: React.PropTypes.object }
+
+  constructor(props: Props, context: Context) {
+    super(props, context);
+    this.uuid = uuid.v4();
+    this.context.JubilationContext = getContext(this.context.JubilationContext);
+    this.setXRange();
+    this.domainMap = getDomainMap(this.uuid, props.data.map(d => ({ x: d, y: 0 })));
+    this.context.JubilationContext.addDomain(this.domainMap);
+  }
+
+  componentWillReceiveProps(nextProps: Props) {
+    this.domainMap = getDomainMap(this.uuid, nextProps.data.map(d => ({ x: d, y: 0 })));
+    this.context.JubilationContext.addDomain(this.domainMap);
+  }
+
+  /**
+   * Inteligently sets the chart's xRange given the passed in label lengths
+   */
+  setXRange(): void {
+    const { xRange } = this.context.JubilationContext;
+    const maxLabelWidth: number = this.props.labels
+      .map(l => getTextWidth(l))
+      .reduce((a, b) => {
+        if (a > b) {
+          return a;
+        }
+        return b;
+      }, 0);
+    this.context.JubilationContext.xRange = [maxLabelWidth + 5, xRange[1]];
+  }
+
+  getHeight() {
+    const binHeight = this.getBinHeight();
+    return binHeight - ((this.props.gapWidth * binHeight) * 2);
+  }
+
+  getBinHeight() {
+    const min: number = this.context.JubilationContext.yScale.range()[0];
+    const max: number = this.context.JubilationContext.yScale.range()[1];
+    const yHeight = max - min;
+    return yHeight / this.props.data.length;
+  }
+
+  getDataBinSize() {
+    return this.context.JubilationContext.yScale.domain()[0] / this.props.data.length;
+  }
+
+  getY(i: number) {
+    const dataBinSize = this.getDataBinSize();
+    return ((i + 1) * dataBinSize) - (dataBinSize * this.props.gapWidth);
+  }
+
+  getLabelPositions() {
+    const { yScale } = this.context.JubilationContext;
+    return getTicks(
```

Comment:
Yeah, I saw that. Depending on the order in which the PRs land, we will have to refactor this on one of the two branches. 

Here's the good news. The tests will freak out because things will break. 👍  for test suites!